src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -49,6 +66,111 @@ public static void FindNatvis(NatvisLoader loader)
+            }
+            else
+            {
+                // NatvisDiagnostic section has not been created, we need to watch for the creation.
+                HostConfigurationSection debuggerSection = configStore.GetCurrentUserDebuggerSection();
+
+                if (debuggerSection != null)
+                {
+                    // We only care about the debugger subkey's keys since we are waiting for the NatvisDiagnostics
+                    // section to be created.
+                    RegistryMonitor rm = new RegistryMonitor(debuggerSection, false);
+
+                    rmw = new RegisterMonitorWrapper(rm);
+
+                    rm.RegChanged += (sender, e) =>
+                    {
+                        HostConfigurationSection checkForSection = configStore.GetNatvisDiagnosticSection();
+
+                        if (checkForSection != null)
+                        {
+                            // NatvisDiagnostic section found. Update the logger
+                            SetNatvisLogLevel(checkForSection);
+
+                            // Remove debugger section tracking 
+                            IDisposable disposable = rmw.CurrentMonitor;
+
+                            // Watch NatvisDiagnostic section
+                            rmw = new RegisterMonitorWrapper(CreateAndStartNatvisDiagnosticMonitor(natvisDiagnosticSection));
+
+                            disposable.Dispose();
+                        }
+                    };
+
+                    rm.Start();
+                }
+            }
+
+
+            return rmw;
+        }
+
+        private static RegistryMonitor CreateAndStartNatvisDiagnosticMonitor(HostConfigurationSection natvisDiagnosticSection)
+        {
+            RegistryMonitor rm = new RegistryMonitor(natvisDiagnosticSection, true);
+
+            rm.RegChanged += (sender, e) =>
+            {
+                SetNatvisLogLevel(natvisDiagnosticSection);
+            };
+
+            rm.Start();
+
+            return rm;
+        }
+
+        private static void SetNatvisLogLevel(HostConfigurationSection natvisDiagnosticSection)
+        {
+            string level = natvisDiagnosticSection.GetValue("Level") as string;
+            if (level != null)
+            {
+                level = level.ToLower(CultureInfo.InvariantCulture);
+            }
+            LogLevel logLevel;
+            switch (level)
+            {
+                case "off":
+                    logLevel = LogLevel.None;
+                    break;
+                case "error":
+                    logLevel = LogLevel.Error;
+                    break;
+                case "warning":
+                    logLevel = LogLevel.Warning;
+                    break;
+                case "verbose":
+                default: // Unknown, default to verbose
+                    logLevel = LogLevel.Verbose;
```

Comment:
>logLevel = LogLevel.Verbose; [](http://example.com/codeflow?start=20&length=28)

If the key is missing, is Verbose the correct default?

Reply:
I think setting it in the middle would be better. Setting it to be Warning